### PR TITLE
Added extensions to prevent the bot from blocking words that end with…

### LINKED ISSUE #801: [BUG] - Issues with detecting extensions
### What bug did you encounter?

The bot doesn't recognize .js .py .html and other code/file extensions and removes them as a part of its monitoring policy. 

### What did you expect to happen?

It shouldn't filter such file extension messages, instead it should have let it remain in the chat.

### Screenshots

image

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by AbhijithGanesh**
… code extensions related to programming

# Pull Request

## Related Issue

Closes #801

## Scope

Did you remember to update the `package.json` version number?

- [❌] Major Version Update: X.\_.\_ (for complete code refactors, or large PRs that adjust most of the codebase)
- [✅ ] Minor Version Update: \_.X.\_ (for the addition of new commands)
- [16.14.1] Patch Version Update: \_.\_.X (for bug fixes _in the code_.)
- [x] No Version Update: \_.\_.\_ (no version update for additions to tests, documentation, or anything that isn't end-user facing.)

## Documentation

For _any_ version updates, please verify if the [documentation page](https://www.beccalyria.com/discord-documentation) needs an update. If it does, please [create an issue there](https://github.com/BeccaLyria/discord-documentation/issues/new?assignees=nhcarrigan&labels=%F0%9F%9A%A6+status%3A+awaiting+triage&template=update.md&title=%5BUPDATE%5D) to ensure it is not forgotten.

- [✅ ] My contribution does NOT require a documentation update.
- [ ] My contribution DOES require a documentation update.

### CODE REVIEW

File: src/listeners/linksListener.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export const linksListener: ListenerInt = {
       }
 
       const linkRegex =
-        /(^|\s+)(([a-z]+:\/\/)?(([a-z0-9-]+\.)+((?!txt|json|css|jsx)[a-z]{3,4}|io|me|gg|to|uk|be|in|eu|ru|us|za)))(:[0-9]{1,5})?(.*\s+|\/|$)/gi;
+        /(^|\s+)(([a-z]+:\/\/)?(([a-z0-9-]+\.)+((?!txt|json|css|jsx|py|js|html|css|scss|ts|rb|cpp|c|h|vue|java|rs|asp|ashx|php|pl|xml|rss)[a-z]{3,4}|io|me|gg|to|uk|be|in|eu|ru|us|za)))(:[0-9]{1,5})?(.*\s+|\/|$)/gi;
```
Comment (nhcarrigan) [inline src/listeners/linksListener.ts:45]:
```suggestion
        /(^|\s+)(([a-z]+:\/\/)?(([a-z0-9-]+\.)+((?!txt|json|css|jsx|html|css|scss|cpp|vue|java|asp|ashx|php|xml|rss)[a-z]{3,4}|io|me|gg|to|uk|be|in|eu|ru|us|za)))(:[0-9]{1,5})?(.*\s+|\/|$)/gi;
```
Bot doesn't trigger on 2-letter domains.

**Review by nhcarrigan — CHANGES_REQUESTED** [review]
A quick tweak. :)

File: src/listeners/linksListener.ts (hunk 1 above)
Comment (AbhijithGanesh) [reply src/listeners/linksListener.ts:45]: How do I add regex for two letter domains ?

Comment (AbhijithGanesh) [reply src/listeners/linksListener.ts]: I have fixed it Nicholas, please check.

File: src/listeners/linksListener.ts [hunk 2]
```diff
@@ -42,7 +42,7 @@ export const linksListener: ListenerInt = {
       }
 
       const linkRegex =
-        /(^|\s+)(([a-z]+:\/\/)?(([a-z0-9-]+\.)+((?!txt|json|css|jsx)[a-z]{3,4}|io|me|gg|to|uk|be|in|eu|ru|us|za)))(:[0-9]{1,5})?(.*\s+|\/|$)/gi;
+        /(^|\s+)(([a-z]+:\/\/)?(([a-z0-9-]+\.)+((?!txt|json|css|jsx|html|rss|csh|act|ash|rby|pyc|java|vue|php|ashx|xml|rss|cpp|scss|vue|class|ogg|wav|wpp|pkg|deb|rpm|tar.gz|zip|dmg|iso|toast|db.sqlite3|csv|mdb|sql|tar|msg|emlx|vcf|bat|apk|exe|gadget|wsf|fnt|font|gif|bmp|icon|jpeg|tiff|asp|aspx|cer|htm|jsp|rss|xhtml|key|pps|ppt|pptx|odp|swift|xlsx|dll|bak|cab|cfg|ini|drv|icns|tmp|cur|avi|flv|mpg|mpeg|swf|mov|doc|docx|pdf|rtx|tex|wpd|)[a-z]{3,4}|io|me|gg|to|uk|be|in|eu|ru|us|za)))(:[0-9]{1,5})?(.*\s+|\/|$)/gi;
```
Comment (nhcarrigan) [inline src/listeners/linksListener.ts]:
```suggestion
        /(^|\s+)(([a-z]+:\/\/)?(([a-z0-9-]+\.)+((?!txt|json|css|jsx|html|rss|csh|act|ash|rby|pyc|java|vue|php|ashx|xml|rss|cpp|scss|vue|ogg|wav|wpp|pkg|deb|rpm|zip|dmg|iso|csv|mdb|sql|tar|msg|emlx|vcf|bat|apk|exe|wsf|fnt|font|gif|bmp|icon|jpeg|tiff|asp|aspx|cer|htm|jsp|rss|key|pps|ppt|pptx|odp|xlsx|dll|bak|cab|cfg|ini|drv|icns|tmp|cur|avi|flv|mpg|mpeg|swf|mov|doc|docx|pdf|rtx|tex|wpd|)[a-z]{3,4}|io|me|gg|to|uk|be|in|eu|ru|us|za)))(:[0-9]{1,5})?(.*\s+|\/|$)/gi;
```

Only TLDs between 3 and 4 characters are flagged, so we can remove some of these.

**Review by nhcarrigan — CHANGES_REQUESTED** [review]

File: src/listeners/linksListener.ts [hunk 3]
```diff
@@ -42,7 +42,7 @@ export const linksListener: ListenerInt = {
       }
 
       const linkRegex =
-        /(^|\s+)(([a-z]+:\/\/)?(([a-z0-9-]+\.)+((?!txt|json|css|jsx)[a-z]{3,4}|io|me|gg|to|uk|be|in|eu|ru|us|za)))(:[0-9]{1,5})?(.*\s+|\/|$)/gi;
+        /(^|\s+)(([a-z]+:\/\/)?(([a-z0-9-]+\.)+((?!txt|json|css|jsx|html|rss|csh|act|ash|rby|pyc|java|vue|php|ashx|xml|rss|cpp|scss|vue|class|ogg|wav|wpp|pkg|deb|rpm|tar.gz|zip|dmg|iso|toast|db.sqlite3|csv|mdb|sql|tar|msg|emlx|vcf|bat|apk|exe|gadget|wsf|fnt|font|gif|bmp|icon|jpeg|tiff|asp|aspx|cer|htm|jsp|rss|xhtml|key|pps|ppt|pptx|odp|swift|xlsx|dll|bak|cab|cfg|ini|drv|icns|tmp|cur|avi|flv|mpg|mpeg|swf|mov|doc|docx|pdf|rtx|tex|wpd|)[a-z]{3,4}|io|me|gg|to|uk|be|in|eu|ru|us|za|py|cs|tn|ll|mk|pl|rc|rb|so|sh)))(:[0-9]{1,5})?(.*\s+|\/|$)/gi;
```
Comment (nhcarrigan) [inline src/listeners/linksListener.ts:45]: To confirm, adding the two-character domains *blocks* them. Are you intending to block additional things?

Comment (AbhijithGanesh) [reply src/listeners/linksListener.ts:45]: oh wait the 2 letter domains are ones that will be blocked
I will reverse them right now.

**Review by nhcarrigan — DISMISSED** [review]
Looks good to me.
Will rebase and merge when I have the time.

File: src/listeners/linksListener.ts [hunk 4]
```diff
@@ -42,7 +42,7 @@ export const linksListener: ListenerInt = {
       }
 
       const linkRegex =
-        /(^|\s+)(([a-z]+:\/\/)?(([a-z0-9-]+\.)+((?!txt|json|css|jsx)[a-z]{3,4}|io|me|gg|to|uk|be|in|eu|ru|us|za)))(:[0-9]{1,5})?(.*\s+|\/|$)/gi;
+        /(^|\s+)(([a-z]+:\/\/)?(([a-z0-9-]+\.)+((?!txt|json|css|jsx|html|rss|csh|act|ash|rby|pyc|java|vue|php|ashx|xml|rss|cpp|scss|vue|cgi|ogg|wav|wpp|pkg|deb|rpm|zip|dmg|iso|csv|mdb|sql|tar|msg|emlx|vcf|bat|apk|exe|wsf|fnt|font|gif|bmp|icon|jpeg|tiff|asp|aspx|cer|htm|jsp|rss|key|pps|ppt|pptx|odp|xlsx|dll|bak|cab|cfg|ini|drv|icns|tmp|cur|avi|flv|mpg|mpeg|swf|mov|doc|docx|pdf|rtx|tex|wpd|)[a-z]{3,4}|io|me|gg|to|uk|be|in|eu|ru|us|za|))(:[0-9]{1,5})?(.*\s+|\/|$)/gi;
```
Comment (nhcarrigan) [inline src/listeners/linksListener.ts:45]:
```suggestion
        /(^|\s+)(([a-z]+:\/\/)?(([a-z0-9-]+\.)+((?!txt|json|css|jsx|html|rss|csh|act|ash|rby|pyc|java|vue|php|ashx|xml|rss|cpp|scss|vue|cgi|ogg|wav|wpp|pkg|deb|rpm|zip|dmg|iso|csv|mdb|sql|tar|msg|emlx|vcf|bat|apk|exe|wsf|fnt|font|gif|bmp|icon|jpeg|tiff|asp|aspx|cer|htm|jsp|rss|key|pps|ppt|pptx|odp|xlsx|dll|bak|cab|cfg|ini|drv|icns|tmp|cur|avi|flv|mpg|mpeg|swf|mov|doc|docx|pdf|rtx|tex|wpd|)[a-z]{3,4}|io|me|gg|to|uk|be|in|eu|ru|us|za))(:[0-9]{1,5})?(.*\s+|\/|$)/gi;
```

**Review by nhcarrigan — APPROVED** [review]

**Comment by nhcarrigan** [comment]
## Congratulations

Congratulations on making your first pull request to our project. We have reviewed your contributions and are happy to accept them.

We hope you enjoyed the contribution process, and encourage you to watch for issues labelled `help wanted` for additional opportunities to contribute.

**Pull request merged**